File: 107.py
from collections import deque

class Solution:
    def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        if root is None:
            return res
        q = deque([root])
        while (s := len(q)) != 0:
            lvl = []
            for _ in range(s):
                node = q.popleft()
                lvl.append(node.val)
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            res.append(lvl.copy())
        return res[::-1]

File: 107.cpp
#include <queue>

struct Solution {
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> res;
        if (root == nullptr){
            return res;
        }
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()){
            int s = q.size();
            vector<int> lvl;
            while (s-- > 0){
                TreeNode *node = q.front();
                q.pop();
                lvl.push_back(node->val);
                if (node->left != nullptr){
                    q.push(node->left);
                }
                if (node->right != nullptr){
                    q.push(node->right);
                }
            }
            res.push_back(lvl);
        }
        ranges::reverse(res);
        return res;
    }
};
